src/cli/format-results-cache.js

```diff
@@ -0,0 +1,115 @@
+"use strict";
+
+// Inspired by LintResultsCache from ESLint
+// https://github.com/eslint/eslint/blob/c2d0a830754b6099a3325e6d3348c3ba983a677a/lib/cli-engine/lint-result-cache.js
+
+const fileEntryCache = require("file-entry-cache");
+const stringify = require("fast-json-stable-stringify");
+// eslint-disable-next-line no-restricted-modules
+const { version: prettierVersion } = require("../index.js");
+const { createHash } = require("./utils.js");
+
+const optionsHashCache = new WeakMap();
+const nodeVersion = process && process.version;
+
+/**
+ * @param {*} options
+ * @returns {string}
+ */
+function getHashOfOptions(options) {
+  if (optionsHashCache.has(options)) {
+    return optionsHashCache.get(options);
+  }
+  const hash = createHash(
+    `${prettierVersion}_${nodeVersion}_${stringify(options)}`
+  );
+  optionsHashCache.set(options, hash);
+  return hash;
+}
+
+/**
+ * @typedef {{ hashOfOptions?: string }} OurMeta
+ * @typedef {import("file-entry-cache").FileDescriptor} FileDescriptor
+ *
+ * @param {import("file-entry-cache").FileDescriptor} fileDescriptor
+ * @returns {FileDescriptor["meta"] & OurMeta}
+ */
+function getMetadataFromFileDescriptor(fileDescriptor) {
+  return fileDescriptor.meta;
+}
+
+const cacheStrategies = ["metadata", "content"];
```

Comment:
You can enable `unicorn/prefer-set-has` rule to use `Set`

Reply:
I've tried to enable the rule, but it doesn't work well. I'll try it later. [3f9ffd7](https://github.com/prettier/prettier/pull/12800/commits/3f9ffd78820821f59159ec2afab8075e76ae469c)

Reply:
Ah, didn't notice there is a `.map` call